### DISCUSSION TITLE
How I can exclude some path from serviceworker?

### DISCUSSION THREAD

**Discussion by gianbia** [nuxt/nuxt#18150] [Questions]
Hi,
I build my Nuxt3 project with "browser" preset and run it on a webserver.
The web service is a Rest app that have some endpoint like this:

[serverUrl]/json/[Endpoints]

When I try to fetch data from this Url the serviceWorker 
return a 404 Nuxt page with this State : 404 Internal Server Error (from service worker).

If I flag Bypass for network in Service Workers config everything work right...

How I can exclude some path from nuxt server handling?

Thanks 
Gianmaria

**Comment by micheee** [comment]
I tried to find a solution for this as well (see nuxt/framework#1777) — I kind of circumvented this by manually changing / overwriting the file during a production build: 

`node_modules/@nuxt/nitro/dist/runtime/entries/service-worker.mjs`

```diff
--- old.mjs	2021-11-24 10:33:10.000000000 +0100
+++ new.mjs	2021-11-24 10:33:38.000000000 +0100
@@ -4,7 +4,7 @@
 const STATIC_ASSETS_BASE = process.env.NUXT_STATIC_BASE + "/" + process.env.NUXT_STATIC_VERSION;
 addEventListener("fetch", (event) => {
   const url = new URL(event.request.url);
-  if (url.pathname.includes(".") && !url.pathname.startsWith(STATIC_ASSETS_BASE) && !url.pathname.startsWith("/api")) {
+  if ((url.pathname.startsWith("/myapi")) || (url.pathname.includes(".") && !url.pathname.startsWith(STATIC_ASSETS_BASE) && !url.pathname.startsWith("/api"))) {
     return;
   }
```

But I deem this to be a very bad and ad-hoc solution ;-) 

I think this might be something better covered by a configurable option, such as `nitro.bypassServiceworker as string[]` — I never contributed to nuxt, but maybe this might be a worthwhile undergoing for a first contribution?
